app/src/main/java/org/stepic/droid/ui/adapters/view_hoders/CourseItemViewHolder.kt

```diff
@@ -207,11 +209,42 @@ class CourseItemViewHolder(
             showJoinButton()
         }
 
-        courseItemMore.visibility = if (showMore) {
-            View.VISIBLE
-        } else {
-            View.GONE
+        val needShowProgress = bindProgressView(course)
+        val needShowRating = bindRatingView(course)
+
+        val showContainer = needShowLearners || needShowProgress || needShowRating
+        coursePropertiesContainer.changeVisibility(showContainer)
+
+        courseItemMore.changeVisibility(showMore)
+    }
+
+    private fun bindProgressView(course: Course): Boolean {
+        val progressPercent: Int? = ProgressUtil.getProgressPercent(course.progressObject)
+        val needShow: Boolean =
+                if (progressPercent != null && progressPercent > 0) {
```

Comment:
It's strange that assing function has some side-effects. Maybe it will be better to write it like:
```kotlin
val needShow = progressPercent != null && progressPercent > 0
if (needShow) {
...
}
```